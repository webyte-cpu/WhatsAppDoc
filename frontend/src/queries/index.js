import { gql } from "@apollo/client";

export const GET_NOTIFICATIONS = gql`
  subscription {
    newNotification {
      title
      description
      isSeen
      isArchived
      createdAt
    }
  }
`;
