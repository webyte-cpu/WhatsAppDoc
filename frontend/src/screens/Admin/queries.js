import { gql } from '@apollo/client';

export const GET_DOCTORS = gql`
  query {
    getAllDoctor{
      uid
      firstName
      middleName
      lastName
      birthdate
      licenceNum
      licenceExp
      licenceImg
      verificationStatus
      specialization
      pushToken
    }
  }
`;

export const UPDATE_DOCTOR = gql`
  mutation updateDoctor($uid:UUID, $verificationStatus: VerificationStatus ){
      updateDoctor(uid: $uid, verificationStatus: $verificationStatus){
          uid
      }
  }
`;