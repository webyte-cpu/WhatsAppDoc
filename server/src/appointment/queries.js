import { gql } from 'apollo-server-express';

export const GET_APPOINTMENT = gql`
  query GetAppointment($uid: UUID!) {
    getAppointment(uid: $uid) {
      uid
      patientUid
      doctorClinicUid
      status
      dateTime
      doctorRemarks
      createdAt
      updatedAt
    }
  }
`
export const GET_ALL_APPOINTMENT = gql`
  query GetAllAppointment {
    getAllAppointment {
      uid
      patientUid
      doctorClinicUid
      status
      dateTime
      doctorRemarks
      createdAt
      updatedAt
    }
  }
`

export const CREATE_APPOINTMENT = gql`
  mutation CreateAppointment(
    $doctorClinicUid: UUID,
    $dateTime: DateTime
  ){
    createAppointment(
      doctorClinicUid: $doctorClinicUid,
      dateTime: $dateTime
    ) {
      uid
      patientUid
      doctorClinicUid
      status
      dateTime
      doctorRemarks
      createdAt
      updatedAt
    }
  }
`

export const UPDATE_APPOINTMENT = gql`
  mutation UpdateAppointment(
    $uid: UUID
    $status: Status
    $dateTime: DateTime
    $doctorRemarks: String
  ){
    updateAppointment(
      uid: $uid
      status: $status
      dateTime: $dateTime
      doctorRemarks: $doctorRemarks
    ) {
      uid
      patientUid
      doctorClinicUid
      status
      dateTime
      doctorRemarks
    }
  }
`