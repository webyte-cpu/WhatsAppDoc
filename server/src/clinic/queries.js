import { gql } from "apollo-server-express";

export const GET_CLINIC = gql`
  query GetClinic($uid: UUID, $doctorUid: UUID) {
    getClinic(uid: $uid, doctorUid: $doctorUid) {
      name
      roomNumber
      address {
        address
        city
        province
        zipCode
        country
      }
      minimumSchedulingNoticeMins
      slotDurationInMins
      consultationFee
    }
  }
`;

export const CREATE_CLINIC = gql`
  mutation CreateClinic(
    $name: String!
    $roomNumber: String!
    $address: AddressInput!
    $minimumSchedulingNoticeMins: Int!
    $slotDurationInMins: Int!
    $consultationFee: Int
  ) {
    createClinic(
      name: $name
      roomNumber: $roomNumber
      address: $address
      minimumSchedulingNoticeMins: $minimumSchedulingNoticeMins
      slotDurationInMins: $slotDurationInMins
      consultationFee: $consultationFee
    ) {
      name
      roomNumber
      address {
        address
        city
        province
        zipCode
        country
      }
      minimumSchedulingNoticeMins
      slotDurationInMins
      consultationFee
    }
  }
`;

export const UPDATE_CLINIC = gql`
  mutation UpdateClinic(
    $uid: UUID!
    $name: String
    $roomNumber: String
    $address: AddressInput
    $minimumSchedulingNoticeMins: Int
    $slotDurationInMins: Int
    $consultationFee: Int
  ) {
    updateClinic(
      uid: $uid
      name: $name
      roomNumber: $roomNumber
      address: $address
      minimumSchedulingNoticeMins: $minimumSchedulingNoticeMins
      slotDurationInMins: $slotDurationInMins
      consultationFee: $consultationFee
    ) {
      uid
      name
      roomNumber
    }
  }
`;

export const DELETE_CLINIC = gql`
  mutation DeleteClinic($uid: UUID!) {
    deleteClinic(uid: $uid) {
      name
      roomNumber
      address {
        address
        city
        province
        zipCode
        country
      }
      minimumSchedulingNoticeMins
      slotDurationInMins
      consultationFee
    }
  }
`;
