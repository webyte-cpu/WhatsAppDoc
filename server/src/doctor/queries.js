import { gql } from "apollo-server-express";

export const VERIFY_DOCTORS = gql`
  mutation updateDoctor($uid: UUID!, $verificationStatus: VerificationStatus) {
    updateDoctor(uid: $uid, verificationStatus: $verificationStatus) {
      uid
    }
  }
`;
