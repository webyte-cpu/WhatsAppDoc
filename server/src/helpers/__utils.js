import objectKeysToCamelCase from "./objectKeyCase.js";
import objectFilter from "./objectFilter.js";
import { constructTestServer, cleanDb ,} from "./integrationTesting.js";

export {
  constructTestServer,
  objectKeysToCamelCase,
  objectFilter,
  cleanDb,
};
