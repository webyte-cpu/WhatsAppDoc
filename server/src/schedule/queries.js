import { gql } from "apollo-server-express";

export const GET_SCHEDULE = gql`
  query getSchedule($uid: UUID!) {
    getSchedule(uid: $uid) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;

export const UPSERT_SCHEDULE = gql`
  mutation upsertSched($doctorClinicUid: UUID!, $schedList: [SchedUpsertInput!]) {
    upsertSchedule(doctorClinicUid: $doctorClinicUid, schedList: $schedList) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;

export const CREATE_SCHEDULE = gql`
  mutation createSched($doctorClinicUid: UUID!, $schedList: [SchedInput!]) {
    createSchedule(doctorClinicUid: $doctorClinicUid, schedList: $schedList) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;

export const UPDATE_SCHEDULE = gql`
  mutation updateSchedule(
    $uid: UUID!
    $startTime: LocalTime
    $endTime: LocalTime
    $daysOfTheWeek: [Int]
  ) {
    updateSchedule(
      uid: $uid
      startTime: $startTime
      endTime: $endTime
      daysOfTheWeek: $daysOfTheWeek
    ) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;

export const DELETE_SCHEDULE = gql`
  mutation deleteSchedules($uids: [UUID!]) {
    deleteSchedules(uids: $uids) {
      startTime
      endTime
      daysOfTheWeek
    }
  }
`;
