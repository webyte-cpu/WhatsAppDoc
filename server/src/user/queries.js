import { gql } from "apollo-server-express";

export const GET_USER = gql`
  query GetUser($uid: UUID!) {
    getUser(uid: $uid) {
      firstName
      lastName
      email
      uid
      role
      updatedAt
    }
  }
`;

export const SIGN_IN = gql`
  mutation SignIn($email: EmailAddress!, $password: Password!) {
    signIn(email: $email, password: $password)
  }
`;

export const SIGN_UP = gql`
  mutation SignUp(
    $firstName: String!
    $middleName: String
    $lastName: String!
    $email: EmailAddress!
    $password: Password!
    $role: Role!
    $sex: Sex!
    $birthdate: Date!
  ) {
    signUp(
      firstName: $firstName
      middleName: $middleName
      lastName: $lastName
      email: $email
      password: $password
      role: $role
      sex: $sex
      birthdate: $birthdate
    )
  }
`;

export const UPDATE_USER = gql`
  mutation UpdateUser(
    $uid: UUID!
    $firstName: String
    $middleName: String
    $lastName: String
    $password: Password
    $role: Role
    $sex: Sex
    $birthdate: Date
    $img: String
  ) {
    updateUser(
      uid: $uid
      firstName: $firstName
      middleName: $middleName
      lastName: $lastName
      password: $password
      role: $role
      sex: $sex
      birthdate: $birthdate
      img: $img
    ) {
      uid
      firstName
      middleName
      lastName
    }
  }
`;

export const DELETE_USER = gql`
  mutation DeleteUser($uid: UUID!) {
    deleteUser(uid: $uid) {
      uid
      firstName
      middleName
      lastName
    }
  }
`;
