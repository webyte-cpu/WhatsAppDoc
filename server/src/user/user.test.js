import { createTestClient } from "apollo-server-testing";
import { cleanDb, constructTestServer } from "../helpers/__utils.js";
import { GET_USER, SIGN_UP } from "./queries.js";

describe("Queries", () => {
  it("fetches single user", async () => {
    // create a test server to test against, using our production typeDefs,
    // resolvers, and dataSources.
    const { server } = constructTestServer();

    // use the test server to create a query function
    const { query } = createTestClient(server);

    // run query against the server and snapshot the output
    const res = await query({
      query: GET_USER,
      variables: { uid: "8b686839-4c94-48ac-ac51-86d36b7bf17c" },
    });
    expect(res).toMatchSnapshot();
  });

  it("it signs up user", async () => {
    afterAll(() => {
      return cleanDb(); // with that we are cleaning the database after all the tests have been executed
    });
    // create a test server to test against, using our production typeDefs,
    // resolvers, and dataSources.
    const { server } = constructTestServer();

    // use the test server to create a query function
    const { query } = createTestClient(server);

    // run query against the server and snapshot the output
    const res = await query({
      query: SIGN_UP,
      variables: {
        firstName: "Kyle",
        lastName: "Osunero",
        email: "kyle2020@webyte.org",
        password: "L3Tm31N",
        role: "ADMIN",
      },
    });
    expect(res).toMatchSnapshot();
  });
});
