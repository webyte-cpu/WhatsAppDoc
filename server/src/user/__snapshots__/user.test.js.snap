// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quieries fetches single user 1`] = `
Object {
  "data": Object {
    "getUser": Object {
      "email": "kyle@webyte.org",
      "firstName": "Kyle",
      "lastName": "Osunero",
      "role": "ADMIN",
      "uid": "8b686839-4c94-48ac-ac51-86d36b7bf17c",
      "updatedAt": null,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Quieries it signs uo user 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: insert into "users" ("created_at", "user_email", "user_first_name", "user_last_name", "user_password", "user_role", "user_uid") values ($1, $2, $3, $4, $5, $6, $7) returning * - duplicate key value violates unique constraint "users_user_email_unique"],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests Queries and Mutations Signs in up 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Unable to acquire a connection],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests Queries and Mutations fetches single user 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Unable to acquire a connection],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Tests Queries and Mutations it signs up user 1`] = `
Object {
  "data": null,
  "errors": Array [
    [GraphQLError: Unable to acquire a connection],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
